controllers/storagecluster/storagecluster_controller.go

```diff
@@ -164,13 +164,17 @@ func (r *StorageClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		},
 	}
 
-	onboardingSecretPredicates := builder.WithPredicates(
-		predicate.NewPredicateFuncs(
-			func(client client.Object) bool {
-				return client.GetName() == onboardingTicketPublicKeySecretName
-			},
-		),
-	)
+	secretPredicate := predicate.Funcs{
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return true
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return e.Object.GetName() != onboardingTicketPublicKeySecretName
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return e.ObjectNew.GetName() != onboardingTicketPublicKeySecretName
+		},
+	}
```

Comment:
@rchikatw Why do we need this? 

If the operator, today or in the future, watches secrets this predicate will filter out these watches for create and update events. That is a bug!

If the underlying reason is that we want to act on events on our secret then we are already doing it with the `Owns` watch so this predicate is not needed anymore.  

If you want to NOT act on private secret events (kind of black listing these events) then you need a different predicate implementation

Reply:
So should we implement blacklisting events on private onboarding secret? or Should i use owns on all secrets like `.Owns(&corev1.Secret{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).` a default predicate

Reply:
I think that blacklisting the private key is an overkill, we should not care if it triggers a reconcile or not as that reconcile will be a noop